enhancements/sig-architecture/64-placementStrategy/README.md

```diff
@@ -301,35 +313,189 @@ status:
+    clusterCount: 150
+  - decisionGroupIndex: 2
+    placementDecisions:
+    - ztp-placement-decision-4
+    clusterCount: 20
+```
+
+The Placement Helper [library](https://github.com/open-cluster-management-io/api/blob/main/cluster/v1beta1/helpers.go) should provide functions to facilitate retrieve the decision groups.
 
 #### Special case handling
 
 * Clusters have been removed from decision group index 1 and decision group index 2 has been created. In order to not shift current selected clusters in the decision groups, placement controller will keep append clusters to decision group index 2 till it reach max number of clusters per decision group then append new cluster to decision group index 1 till it reach the max number of clusters per decision group.
 
+* The PrioritizerPolicy has been defined by end user and a new cluster has been added to the placement selection after the decision Groups are created. The clusters wil need to be adjusted within the decision groups based on the new cluster decision group index.
 
 ### How Workload applier APIs will benefits from using placementStrategy
 
-The OCM APIs such as ManifestWorkReplicaSet or Policy can define how-to roll out the workloads considering the Decision Groups defined in the placement.
-For example; assuming the below placementStrategy struct is part of the manifestWorkReplicaSet APIs. It defines the workload rollout type (All or Progressive) and decision groups to apply the workload first.
+The OCM APIs such as ManifestWorkReplicaSet or Policy can define how-to roll out the workloads considering the Decision Groups defined in the placement. The RolloutStrategy struct proposed below provide a common APIs that can be consumed by workload applier APIs to rollout the workload. 
 
 ```go
-type PlacementStrategy struct {
-	// Rollout type either All or Progressive.
-	// +kubebuilder:validation:Enum=All;Progressive
+// RolloutStrategy Type
+type RolloutStrategyType string
+
+const (
+	//All means apply the workload to all clusters in the decision groups at once.
+	All RolloutStrategyType = "All"
+	//Progressive means apply the workload to the selected clusters progressively per cluster.
+	Progressive RolloutStrategyType = "Progressive"
+	//ProgressivePerGroup means apply the workload to the selected clusters progressively per group.
+	ProgressivePerGroup RolloutStrategyType = "ProgressivePerGroup"
+)
+
+// MandatoryDecisionGroup set the decision group name and group index.
+type MandatoryDecisionGroup struct {
+	// GroupName of the decision group should match the placementDecisions label value with label key cluster.open-cluster-management.io/decision-group-name
+	// +optional
+	GroupName string `json:"groupName,omitempty"`
+
+	// GroupIndex of the decision group should match the placementDecisions label value with label key cluster.open-cluster-management.io/decision-group-index
+	// +optional
+	GroupIndex int32 `json:"groupIndex,omitempty"`
+}
+
+type RolloutStrategy struct {
```

Comment:
How about 
```
type RolloutStrategy struct {
	// +kubebuilder:default:=All
	// +optional
	Type RolloutStrategyType `json:"type,omitempty"`
	
	// +optional
	All *RolloutAll `json:”all,omitempty"`

	// +optional
	Progressive *RolloutProgressive `json:”progressive,omitempty"`

	// +optional
	ProgressivePerGroup *RolloutProgressivePerGroup `json:”progressivePerGroup,omitempty"`
}

type RolloutAll struct {
// list the supported field for each type
// ...
}

type RolloutProgressive struct {
// ...
}

type RolloutProgressivePerGroup struct {
// ...
}

Reply:
Well, the RolloutStrategyType defined as string type in order to give more flexibility for consumer APIs. For example a consumer APIs (controller) need a different rolloutType than All, Progressive and ProgressivePerGroup.  So it can define a new rolloutType in the consumer API and use it without affecting the commonly used rolloutStrategy.